src/config.h

```diff
@@ -27,129 +27,91 @@ std::ostream& operator<<(std::ostream& os, CryptoSource cs);
 // updateFromPropertyTree() in config.cc.
 
 struct GatewayConfig {
-  GatewayConfig() : http(true), socket(false) {}
-  bool http;
-  bool socket;
+  bool socket{false};
 };
 
 struct NetworkConfig {
-  NetworkConfig() : socket_commands_path("/tmp/sota-commands.socket"), socket_events_path("/tmp/sota-events.socket") {
-    socket_events.push_back("DownloadComplete");
-    socket_events.push_back("DownloadFailed");
-  }
-  std::string socket_commands_path;
-  std::string socket_events_path;
-  std::vector<std::string> socket_events;
+  std::string socket_commands_path{"/tmp/sota-commands.socket"};
+  std::string socket_events_path{"/tmp/sota-events.socket"};
+  std::vector<std::string> socket_events{"DownloadComplete", "DownloadFailed"};
 };
 
 struct P11Config {
-  P11Config() {}
-  boost::filesystem::path module;
-  std::string pass;
-  std::string uptane_key_id;
-  std::string tls_cacert_id;
-  std::string tls_pkey_id;
-  std::string tls_clientcert_id;
+  boost::filesystem::path module{};
+  std::string pass{};
+  std::string uptane_key_id{};
+  std::string tls_cacert_id{};
+  std::string tls_pkey_id{};
+  std::string tls_clientcert_id{};
 };
 
-class TlsConfig {
- public:
-  TlsConfig() : server(""), server_url_path(""), ca_source(kFile), pkey_source(kFile), cert_source(kFile) {}
-
-  std::string server;
-  boost::filesystem::path server_url_path;
-  CryptoSource ca_source;
-  CryptoSource pkey_source;
-  CryptoSource cert_source;
+struct TlsConfig {
+  std::string server{};
+  boost::filesystem::path server_url_path{};
+  CryptoSource ca_source{kFile};
+  CryptoSource pkey_source{kFile};
+  CryptoSource cert_source{kFile};
 };
 
 struct ProvisionConfig {
-  ProvisionConfig() : server(""), p12_password(""), expiry_days("36000"), provision_path(""), mode(kAutomatic) {}
-  std::string server;
-  std::string p12_password;
-  std::string expiry_days;
-  boost::filesystem::path provision_path;
-  ProvisionMode mode;
+  std::string server{};
```

Comment:
Default constuctor for server will automatically called on ProvisionConfig construction. So I think it would be nicer to remove empty {}

Reply:
Yes, I kept them for consistency, because it's easy to forget the braces for integers and other primitive types. But I don't feel strongly about this, so we can omit them for non-primitive types.